Automattic/wp-calypso
# chore: Clean up unused `client` dependencies

#### Changes proposed in this Pull Request

Removes unused dependencies in `client/package.json`.

To find them, I run `yarn dlx depcheck` first to get the list of unused dependencies. And for each unused dependency I searched its name in `client/**/*` to make sure it is actually not used (there were a few false positives btw).

To verify it works, I built this branch with `CALYPSO_ENV=production NODE_ENV=production MINIFY_JS=false yarn build` and diffed the directory `./public` with a built from `trunk`:

```
$ diff -qr wp-calypso/public wp-calypso2/public                   
                                                                    
Only in wp-calypso/public/evergreen: 59749.0f14d460bb68fad62fcc.min.js
Only in wp-calypso2/public/evergreen: 59749.1d0b3057da7801cc7e38.min.js
Only in wp-calypso/public/evergreen: runtime.79633ec6f752b394873a.min.js
Only in wp-calypso2/public/evergreen: runtime.bb613cabb91ed3376ce1.min.js
```

The differences can be easily explained and are innocuous:

* The diff in `59749.*.js` is related to the `immutable` path, `../node_modules/immutable` vs `../node_modules/draft-js/node_modules/immutable`. As we don't depend on `immutable` anymore, the version provided by `draft-js` can be hoisted to the root `node_modules`, hence the difference

* The diff in `runtime.*.js` is caused because the `59749.*.js` files have changed their name.


#### Testing instructions

* Verify all tests are green
* Check the brach, build it with `CALYPSO_ENV=production NODE_ENV=production MINIFY_JS=false yarn build` and diff the results with `trunk`.
